Observer/InvoiceSaveAfter.php

```diff
@@ -40,6 +47,19 @@ public function __construct(
     public function execute(\Magento\Framework\Event\Observer $observer)
     {
         try {
+            $extensionAttributes = $observer->getInvoice()->getExtensionAttributes();
+            if ($extensionAttributes === null) {
+                $extensionAttributes = $this->invoiceExtensionFactory->create();
+            }
+            $shippingProtection = $extensionAttributes->getShippingProtection();
+
+            if ($observer->getInvoice() && $shippingProtection) {
```

Comment:
Do we need the invoice check here? Isn't the presence of `$shippingProtection` enough?